src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -910,5 +898,49 @@ private bool IsFailOverable(RequestFailedException rfe)
                    rfe.Status == (int)HttpStatusCode.RequestTimeout ||
                    rfe.Status >= (int)HttpStatusCode.InternalServerError;
         }
+
+        private async Task<Dictionary<string, ConfigurationSetting>> MapConfigurationSettings(Dictionary<string, ConfigurationSetting> data)
+        {
+            Dictionary<string, ConfigurationSetting> mappedData = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (KeyValuePair<string, ConfigurationSetting> kvp in data)
+            {
+                ConfigurationSetting setting = kvp.Value;
+                foreach (Func<ConfigurationSetting, ValueTask<ConfigurationSetting>> func in _options.Mappers)
+                {
+                    setting = await func(setting).ConfigureAwait(false);
+                }
+                if (setting != null)
+                {
+                    mappedData[kvp.Key] = setting;
+                }
+            }
+
+            return mappedData;
+        }
+
+        private IEnumerable<ConfigurationSetting> GetCurrentKeyValueCollection(string key, string label, IEnumerable<ConfigurationSetting> existingSettings)
+        {
+            IEnumerable<ConfigurationSetting> currentKeyValues;
+
+            if (key.EndsWith("*"))

```

Comment:
What's the case where we won't end up with it ending in '*'?

Reply:
From what I understand, it's when a key value is specified in the select method for feature flags options

Reply:
I see now. Unfortunate that we lump in single key refresh with collection based refresh. I'd like if we could rework the single key case to somehow use the single key refresh pipeline. That's out of the scope for this PR though.